server/game/cards/20-HMW/BraveCompanions.js

```diff
@@ -0,0 +1,58 @@
+const DrawCard = require('../../drawcard.js');
+const {Tokens} = require('../../Constants');
+
+class BraveCompanions extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            match: card => card === this,
+            effect: ability.effects.dynamicStrength(() => this.calculateStrength())
+        });
+                      
+        this.reaction({
+            when: {
+                onCardEntersPlay: event => event.card === this && event.playingType === 'marshal'
+            },
+            handler: () => {
+                this.game.promptForDeckSearch(this.controller, {
```

Comment:
I think it should be possible to implement this one using the new style game actions. See Benjen's Cache for an example.

Reply:
fixed